liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -277,11 +299,62 @@ private void addGenerationCommentToConfig(ServerConfigXmlDocument doc, File serv
 
                 String eeVersion = getEEVersion(project); 
                 String mpVersion = getMPVersion(project);
-                List<String> currentFeatures = new ArrayList<String>(currentFeatureSet);
+                List<String> currentFeatures;
+                if (currentFeatureSet == null) { // signifies we are calling the binary scanner for a sample list of features
+                    currentFeatures = new ArrayList<String>();
+                } else {
+                    currentFeatures = new ArrayList<String>(currentFeatureSet);
+                }
                 log.debug("The following messages are from the application binary scanner used to generate Liberty features");
                 featureList = (Set<String>) driveScanMavenFeatureList.invoke(null, binaryInputs, eeVersion, mpVersion, currentFeatures, java.util.Locale.getDefault());
                 log.debug("End of messages from application binary scanner. Features recommended :");
                 for (String s : featureList) {log.debug(s);};
+            } catch (InvocationTargetException ite) {
+                // This is the exception from the JVM that indicates there was an exception in the method we
+                // called through reflection. We must extract the actual exception from the 'cause' field.
+                // A RuntimeException means the currentFeatureSet contains conflicts.
+                // A FeatureConflictException means the binary files scanned conflict with each other or with
+                // the currentFeatureSet parameter.
+                Throwable scannerException = ite.getCause();
+                if (scannerException instanceof RuntimeException) {
+                    // The list of features from the app is passed in but it contains conflicts 
+                    String problemMessage = scannerException.getMessage();
+                    if (problemMessage == null || problemMessage.isEmpty()) {
+                        log.debug("RuntimeException from binary scanner without descriptive message", scannerException);
```

Comment:
Since we don't know what might cause this I thought to exit silently but I think it is better if we display an error before exiting generate-features. The user will notice there are no new features.